// ngx-skill-bar/esm2015/lib/ngx-skill-bar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from "@angular/core";
import { Bar } from "./bar.model";
export class NgxSkillBarComponent {
    constructor() {
        this.percentage = 0;
        this.numberOfBars = 5;
        this.standarWidthBar = 20;
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        this.init();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.init();
    }
    /**
     * @return {?}
     */
    init() {
        this.standarWidthBar = 100 / this.numberOfBars;
        this.bars = [...this.generateLine()];
    }
    /**
     * @return {?}
     */
    generateLine() {
        let /** @type {?} */ bars = [];
        for (let /** @type {?} */ i = 1; i <= this.numberOfBars; i++) {
            bars = [...bars, new Bar(this.calculateBar(i))];
        }
        return bars;
    }
    /**
     * @param {?} indexBar
     * @return {?}
     */
    calculateBar(indexBar) {
        const /** @type {?} */ substraction = this.percentage - (indexBar - 1) * this.standarWidthBar;
        if (this.percentage >= indexBar * this.standarWidthBar) {
            return 100;
        }
        else if (substraction > 0) {
            return (substraction * 100) / this.standarWidthBar;
        }
        return 0;
    }
}
NgxSkillBarComponent.decorators = [
    { type: Component, args: [{
                selector: "ngx-skill-bar",
                template: `
    <statistics-list-bar [full]="full" [barWidth]="barWidth" [showFigures]="showFigures" [percentage]="percentage" [bars]="bars" [bgColor]="bgColor" [color]="color"></statistics-list-bar>
  `,
                styles: []
            },] },
];
/** @nocollapse */
NgxSkillBarComponent.ctorParameters = () => [];
NgxSkillBarComponent.propDecorators = {
    percentage: [{ type: Input }],
    numberOfBars: [{ type: Input }],
    showFigures: [{ type: Input }],
    bgColor: [{ type: Input }],
    color: [{ type: Input }],
    barWidth: [{ type: Input }],
    full: [{ type: Input }]
};
function NgxSkillBarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxSkillBarComponent.prototype.percentage;
    /** @type {?} */
    NgxSkillBarComponent.prototype.numberOfBars;
    /** @type {?} */
    NgxSkillBarComponent.prototype.showFigures;
    /** @type {?} */
    NgxSkillBarComponent.prototype.bgColor;
    /** @type {?} */
    NgxSkillBarComponent.prototype.color;
    /** @type {?} */
    NgxSkillBarComponent.prototype.barWidth;
    /** @type {?} */
    NgxSkillBarComponent.prototype.full;
    /** @type {?} */
    NgxSkillBarComponent.prototype.standarWidthBar;
    /** @type {?} */
    NgxSkillBarComponent.prototype.bars;
}
