// ngx-skill-bar/esm5/lib/statistics-list-bar/statistics-list-bar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from "@angular/core";
var /** @type {?} */ DEFAULT_PADDING = 2;
var StatisticsListBarComponent = /** @class */ (function () {
    function StatisticsListBarComponent() {
        this.bars = [];
        this.showFigures = false;
        this.percentage = 0;
        this.full = false;
        this.padding = DEFAULT_PADDING;
    }
    /**
     * @return {?}
     */
    StatisticsListBarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.full == true) {
            this.padding = 0;
        }
        else {
            this.padding = DEFAULT_PADDING;
        }
    };
    /**
     * @return {?}
     */
    StatisticsListBarComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () { };
    StatisticsListBarComponent.decorators = [
        { type: Component, args: [{
                    selector: "statistics-list-bar",
                    template: "<div class=\"statistics-container\">\n    <statistics-bar \n        class=\"statistics-item\" \n        *ngFor=\"let bar of bars\"  \n        [width]=\"bar.width\" \n        [bgColor]=\"bgColor\" \n        [color]=\"color\" \n        [barWidth]=\"barWidth\"\n        [ngStyle]=\"{'padding': padding+'px'}\"\n        ></statistics-bar>\n    <div class=\"figures\" *ngIf=\"showFigures\">{{percentage}}%</div>\n</div>\n",
                    styles: [".statistics-container{display:flex;flex-wrap:wrap}.statistics-item{margin-top:1px;padding:2px}.figures{margin-top:-4px;margin-left:4px}"]
                },] },
    ];
    /** @nocollapse */
    StatisticsListBarComponent.ctorParameters = function () { return []; };
    StatisticsListBarComponent.propDecorators = {
        bars: [{ type: Input }],
        showFigures: [{ type: Input }],
        percentage: [{ type: Input }],
        bgColor: [{ type: Input }],
        color: [{ type: Input }],
        barWidth: [{ type: Input }],
        full: [{ type: Input }]
    };
    return StatisticsListBarComponent;
}());
export { StatisticsListBarComponent };
function StatisticsListBarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    StatisticsListBarComponent.prototype.bars;
    /** @type {?} */
    StatisticsListBarComponent.prototype.showFigures;
    /** @type {?} */
    StatisticsListBarComponent.prototype.percentage;
    /** @type {?} */
    StatisticsListBarComponent.prototype.bgColor;
    /** @type {?} */
    StatisticsListBarComponent.prototype.color;
    /** @type {?} */
    StatisticsListBarComponent.prototype.barWidth;
    /** @type {?} */
    StatisticsListBarComponent.prototype.full;
    /** @type {?} */
    StatisticsListBarComponent.prototype.padding;
}
