// ngx-skill-bar/esm2015/lib/statistics-bar/statistics-bar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ChangeDetectorRef, Input } from "@angular/core";
export class StatisticsBarComponent {
    /**
     * @param {?} cdr
     */
    constructor(cdr) {
        this.cdr = cdr;
        this.width = 0;
        this.bgColor = "#644d4d";
        this.color = "white";
        this.barWidth = 40;
    }
    /**
     * @return {?}
     */
    ngOnInit() { }
    /**
     * @return {?}
     */
    ngOnChanges() { }
}
StatisticsBarComponent.decorators = [
    { type: Component, args: [{
                selector: "statistics-bar",
                template: `<div class="bar" 
  [ngStyle]="{'background-color': bgColor, 'width': barWidth+'px'}">
  <div [ngStyle]="{'width': width+'%', 'background-color': color}"></div>
</div>`,
                styles: [`.bar{background-color:#644d4d;height:9px;width:40px}.bar>div{background-color:#fff;height:9px}`]
            },] },
];
/** @nocollapse */
StatisticsBarComponent.ctorParameters = () => [
    { type: ChangeDetectorRef }
];
StatisticsBarComponent.propDecorators = {
    width: [{ type: Input }],
    bgColor: [{ type: Input }],
    color: [{ type: Input }],
    barWidth: [{ type: Input }]
};
function StatisticsBarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    StatisticsBarComponent.prototype.width;
    /** @type {?} */
    StatisticsBarComponent.prototype.bgColor;
    /** @type {?} */
    StatisticsBarComponent.prototype.color;
    /** @type {?} */
    StatisticsBarComponent.prototype.barWidth;
    /** @type {?} */
    StatisticsBarComponent.prototype.cdr;
}
