// ngx-skill-bar/esm2015/lib/statistics-list-bar/statistics-list-bar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from "@angular/core";
const /** @type {?} */ DEFAULT_PADDING = 2;
export class StatisticsListBarComponent {
    constructor() {
        this.bars = [];
        this.showFigures = false;
        this.percentage = 0;
        this.full = false;
        this.padding = DEFAULT_PADDING;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (this.full == true) {
            this.padding = 0;
        }
        else {
            this.padding = DEFAULT_PADDING;
        }
    }
    /**
     * @return {?}
     */
    ngOnChanges() { }
}
StatisticsListBarComponent.decorators = [
    { type: Component, args: [{
                selector: "statistics-list-bar",
                template: `<div class="statistics-container">
    <statistics-bar 
        class="statistics-item" 
        *ngFor="let bar of bars"  
        [width]="bar.width" 
        [bgColor]="bgColor" 
        [color]="color" 
        [barWidth]="barWidth"
        [ngStyle]="{'padding': padding+'px'}"
        ></statistics-bar>
    <div class="figures" *ngIf="showFigures">{{percentage}}%</div>
</div>
`,
                styles: [`.statistics-container{display:flex;flex-wrap:wrap}.statistics-item{margin-top:1px;padding:2px}.figures{margin-top:-4px;margin-left:4px}`]
            },] },
];
/** @nocollapse */
StatisticsListBarComponent.ctorParameters = () => [];
StatisticsListBarComponent.propDecorators = {
    bars: [{ type: Input }],
    showFigures: [{ type: Input }],
    percentage: [{ type: Input }],
    bgColor: [{ type: Input }],
    color: [{ type: Input }],
    barWidth: [{ type: Input }],
    full: [{ type: Input }]
};
function StatisticsListBarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    StatisticsListBarComponent.prototype.bars;
    /** @type {?} */
    StatisticsListBarComponent.prototype.showFigures;
    /** @type {?} */
    StatisticsListBarComponent.prototype.percentage;
    /** @type {?} */
    StatisticsListBarComponent.prototype.bgColor;
    /** @type {?} */
    StatisticsListBarComponent.prototype.color;
    /** @type {?} */
    StatisticsListBarComponent.prototype.barWidth;
    /** @type {?} */
    StatisticsListBarComponent.prototype.full;
    /** @type {?} */
    StatisticsListBarComponent.prototype.padding;
}
