// ngx-skill-bar/esm5/lib/ngx-skill-bar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from "@angular/core";
import { Bar } from "./bar.model";
var NgxSkillBarComponent = /** @class */ (function () {
    function NgxSkillBarComponent() {
        this.percentage = 0;
        this.numberOfBars = 5;
        this.standarWidthBar = 20;
    }
    /**
     * @return {?}
     */
    NgxSkillBarComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        this.init();
    };
    /**
     * @return {?}
     */
    NgxSkillBarComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.init();
    };
    /**
     * @return {?}
     */
    NgxSkillBarComponent.prototype.init = /**
     * @return {?}
     */
    function () {
        this.standarWidthBar = 100 / this.numberOfBars;
        this.bars = tslib_1.__spread(this.generateLine());
    };
    /**
     * @return {?}
     */
    NgxSkillBarComponent.prototype.generateLine = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ bars = [];
        for (var /** @type {?} */ i = 1; i <= this.numberOfBars; i++) {
            bars = tslib_1.__spread(bars, [new Bar(this.calculateBar(i))]);
        }
        return bars;
    };
    /**
     * @param {?} indexBar
     * @return {?}
     */
    NgxSkillBarComponent.prototype.calculateBar = /**
     * @param {?} indexBar
     * @return {?}
     */
    function (indexBar) {
        var /** @type {?} */ substraction = this.percentage - (indexBar - 1) * this.standarWidthBar;
        if (this.percentage >= indexBar * this.standarWidthBar) {
            return 100;
        }
        else if (substraction > 0) {
            return (substraction * 100) / this.standarWidthBar;
        }
        return 0;
    };
    NgxSkillBarComponent.decorators = [
        { type: Component, args: [{
                    selector: "ngx-skill-bar",
                    template: "\n    <statistics-list-bar [full]=\"full\" [barWidth]=\"barWidth\" [showFigures]=\"showFigures\" [percentage]=\"percentage\" [bars]=\"bars\" [bgColor]=\"bgColor\" [color]=\"color\"></statistics-list-bar>\n  ",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    NgxSkillBarComponent.ctorParameters = function () { return []; };
    NgxSkillBarComponent.propDecorators = {
        percentage: [{ type: Input }],
        numberOfBars: [{ type: Input }],
        showFigures: [{ type: Input }],
        bgColor: [{ type: Input }],
        color: [{ type: Input }],
        barWidth: [{ type: Input }],
        full: [{ type: Input }]
    };
    return NgxSkillBarComponent;
}());
export { NgxSkillBarComponent };
function NgxSkillBarComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    NgxSkillBarComponent.prototype.percentage;
    /** @type {?} */
    NgxSkillBarComponent.prototype.numberOfBars;
    /** @type {?} */
    NgxSkillBarComponent.prototype.showFigures;
    /** @type {?} */
    NgxSkillBarComponent.prototype.bgColor;
    /** @type {?} */
    NgxSkillBarComponent.prototype.color;
    /** @type {?} */
    NgxSkillBarComponent.prototype.barWidth;
    /** @type {?} */
    NgxSkillBarComponent.prototype.full;
    /** @type {?} */
    NgxSkillBarComponent.prototype.standarWidthBar;
    /** @type {?} */
    NgxSkillBarComponent.prototype.bars;
}
